Tests/PerformanceTest/XCTestCase+InstructionsCount.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import _InstructionCounter
+
+extension XCTestCase {
+  private var baselineURL: URL {
+    if let baselineFile = ProcessInfo.processInfo.environment["BASELINE_FILE"] {
+      return URL(fileURLWithPath: baselineFile)
+    } else {
+      return URL(fileURLWithPath: #file)
+        .deletingLastPathComponent()
+        .appendingPathComponent("baselines.json")
+    }
+  }
+
+  func measureInstructions(_ baselineName: StaticString = #function, block: () -> Void, file: StaticString = #file, line: UInt = #line) throws {
+    let strippedBaselineName = "\(baselineName)".replacingOccurrences(of: "()", with: "")
+    var baseline: UInt64? = try baseline(for: strippedBaselineName)
```

Comment:
I think if 
- No baseline *file* exists, the test should execute `block` (to check that it works functionally) but not emit a failure about the missing baseline. This is so you don’t immediately get hit with test failures if you clone the repo and run the tests
- If a baseline *file* exists but no baseline for this test case, you have opted into performance testing. In that case, there should be a test failure about the missing baseline.